source/MaterialXCore/MaterialNode.h

```diff
@@ -0,0 +1,46 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_MATERIALNODE_H
+#define MATERIALX_MATERIALNODE_H
+
+/// @file
+/// Material node utilities 
+
+#include <MaterialXCore/Library.h>
+
+#include <MaterialXCore/Document.h>
+#include <MaterialXCore/Element.h>
+#include <MaterialXCore/Interface.h>
+
+#include <list>
+
+namespace MaterialX
+{
+
+/// Return a vector of all Shader nodes for a Material node.
+/// @param materialNode Node to example
+/// @param shaderType Type of shader to return. If an empty string is specified then
+///        all shader node types are returned. The default argument value is an empetyr
```

Comment:
Nitpick: The last sentence could be removed, as it's saying that same thing as the first ;)